gusto/diagnostics.py

```diff
@@ -705,7 +705,7 @@ def setup(self, state, vorticity_type=None):
                 gradperp = lambda psi: cross(cell_normals, grad(psi))
                 L = (- inner(gradperp(gamma), u))*dx
             else:
-                raise NotImplementedError("The vorticity diagnostics have only been implemented for 2D spherical geometries.")
+                L = (- inner(state.perp(grad(gamma)), u))*dx
```

Comment:
You should be able to remove the code in the `if` block because you're getting the `perp` function from `state` so it is already correct on the sphere.

Reply:
P.S. thanks for fixing this - I should've done it this way in the first place!

Reply:
No problem at all! `if` statement removed.

Reply:
Thanks! Unfortunately the code cannot currently be built because lint is failing. Lint checks that the code syntax is correct and in this case it is complaining because some functions are imported from Firedrake that are no longer used because the code in the `if` statement was the only place they were used in this file. You can check this locally by running `make lint` - let me know if you have trouble figuring it out...

Reply:
Oh, but the good news is that lint is run after the tests - so the tests all still pass after this change. `test_sw_triangle.py` compares the vorticity to an analytical expression for the vorticity so the computation still works on the sphere - the hard part is done!

Reply:
Removing the unnecessary imports and a bit of white-space locally made lint pass. Would that be an acceptable solution?

Reply:
sounds perfect... expecting the tests to pass very soon then I will merge your pull request - many thanks!